slack/slack/weather/rust/src/weather.rs

```diff
@@ -0,0 +1,61 @@
+use wasmedge_bindgen_macro::*;
+
+use http_req::request;
+use serde_json::Value;
+
+#[wasmedge_bindgen]
+pub fn run(s: String) -> Result<String, String> {
+    #[allow(unused_imports)]
+    use wasmedge_bindgen::*;
+    _run(s)
+}
+
+pub fn _run(s: String) -> Result<String, String> {
+    if s.trim().is_empty() {
+        return Err("Wrong location submitted".to_string());
+    }
+
+    get_weather(&s)
+}
+
+pub fn get_weather(city: &String) -> Result<String, String> {
+    let mut writer = Vec::new();
+    let API_key = "d7708b2a44c24775d4845c07a994e7a0";
+    let query_str = format!(
+        "https://api.openweathermap.org/data/2.5/weather?q={city}&units=metric&appid={API_key}"
+    );
+
+    let _ = request::get(query_str, &mut writer).unwrap();
+
+    let payload: Value = serde_json::from_str(&String::from_utf8(writer).unwrap()).unwrap();
+    match payload.get("cod").unwrap() == "404" {
```

Comment:
It seems like a typo.

Reply:
"cod" isn't a typo, it's openweathermap.com's field name for status code:
{"cod":"404","message":"city not found"}